migration-engine/cli/tests/cli_tests.rs

```diff
@@ -355,6 +356,147 @@ fn introspect_empty_database() {
+                "schema": &schema_path,
+            },
+            "script": &script,
+        }
+    }))
+    .unwrap();
+    stdin.write_all(msg.as_bytes()).unwrap();
+    stdin.write_all(b"\n").unwrap();
+
+    let mut response = String::new();
+    stdout.read_line(&mut response).unwrap();
+
+    assert!(response.starts_with(r##"{"jsonrpc":"2.0","result":null,"id":1}"##));
+}
+
+#[test_connector(tags(Postgres))]
+fn introspect_postgres(api: TestApi) {
+    /* Drop and create database via `drop-database` and `create-database` */
+
+    let connection_string = api.connection_string();
+    let output = api.run(&["--datasource", &connection_string, "drop-database"]);
+    assert!(output.status.success(), "{:#?}", output);
+    let output = api.run(&["--datasource", &connection_string, "create-database"]);
+    assert!(output.status.success(), "{:#?}", output);
+
+    use std::io::{BufRead, BufReader, Write as _};
+    let tmpdir = tempfile::tempdir().unwrap();
+    let schema = r#"
+        datasource db {
+            provider = "postgres"
+            url = env("TEST_DATABASE_URL")
+        }
+    "#;
+    let schema_path = tmpdir.path().join("prisma.schema");
+    std::fs::write(&schema_path, schema).unwrap();
+
+    let mut process = Command::new(migration_engine_bin_path())
+        .stdin(std::process::Stdio::piped())
+        .stdout(std::process::Stdio::piped())
+        .stderr(std::process::Stdio::null())
+        .spawn()
+        .unwrap();
+
+    let stdin = process.stdin.as_mut().unwrap();
+    let mut stdout = BufReader::new(process.stdout.as_mut().unwrap());
+
+    /* Create table via `dbExecute` */
+
+    let script = formatdoc! {r#"
+        CREATE TABLE "public"."A" (
+            id SERIAL PRIMARY KEY,
+            data TEXT
+        );
+    "#};
+    let msg = serde_json::to_string(&serde_json::json!({
+        "jsonrpc": "2.0",
+        "method": "dbExecute",
+        "id": 1,
+        "params": {
+            "datasourceType": {
+                "tag": "schema",
+                "schema": &schema_path,
+            },
+            "script": &script,
+        }
+    }))
+    .unwrap();
+    stdin.write_all(msg.as_bytes()).unwrap();
+    stdin.write_all(b"\n").unwrap();
+
+    let mut response = String::new();
+    stdout.read_line(&mut response).unwrap();
+
+    assert!(response.starts_with(r##"{"jsonrpc":"2.0","result":null,"id":1}"##));
+
+    /* Introspect via `introspect` */
+    let msg = serde_json::to_string(&serde_json::json!({
+        "jsonrpc": "2.0",
+        "method": "introspect",
+        "id": 1,
+        "params": {
+            "schema": &schema,
+            "force": true,
+            "compositeTypeDepth": 5,
+        }
+    }))
+    .unwrap();
+    stdin.write_all(msg.as_bytes()).unwrap();
+    stdin.write_all(b"\n").unwrap();
+
+    let mut response = String::new();
+    stdout.read_line(&mut response).unwrap();
+
+    assert!(response.starts_with(r##"{"jsonrpc":"2.0","result""##));
```

Comment:
Could we use `expect_test` snapshot tests here? Would be easier to fix if we change something, and the test output is nicer. 

example: https://github.com/prisma/prisma-engines/blob/main/introspection-engine/introspection-engine-tests/tests/postgres/gin.rs#L38-L49